doc/reference/pipeline_spec.md

```diff
@@ -186,41 +179,109 @@ avoid scheduling problems that prevent users from being unable to run
+which case `"master"` will be used.
+
+`input.atom.commit` is the `repo` and `branch` (specified as `id`) to be used for the
+input, `repo` is required but `id` may be left blank in which case `"master"`
+will be used.
+
+`input.atom.glob` is a glob pattern that's used to determine how the input data
 is partitioned.  It's explained in detail in the next section.
 
-`inputs.lazy` controls how the data is exposed to jobs. The default is
-`false` which means the job will eagerly download the data it needs to
-process and it will be exposed as normal files on disk. If lazy is set
-to `true`, data will be exposed as named pipes instead and no data will
-be downloaded until the job opens the pipe and reads it, if the pipe is
-never opened then no will be downloaded. Some applications won't work with
-pipes, for example if they make syscalls such as `Seek` which pipes don't
-support. Applications that can work with pipes should use them since they're
-more performant, the difference will be especially notable if the job only
-reads a subset of the files that are available to it.  Note that `lazy`
-currently doesn't support datums that contain more than 10000 files.
-
-`inputs.from` specifies the starting point of the input branch.  If `from`
-is not specified, then the entire input branch will be processed.  Otherwise,
-only commits since the `from` commit (not including the `from` commit itself)
- will be processed.
+`input.atom.lazy` controls how the data is exposed to jobs. The default is `false`
+which means the job will eagerly download the data it needs to process and it
+will be exposed as normal files on disk. If lazy is set to `true`, data will be
+exposed as named pipes instead and no data will be downloaded until the job
+opens the pipe and reads it, if the pipe is never opened then no data will be
+downloaded. Some applications won't work with pipes, for example if they make
+syscalls such as `Seek` which pipes don't support. Applications that can work
+with pipes should use them since they're more performant, the difference will
+be especially notable if the job only reads a subset of the files that are
+available to it.  Note that `lazy` currently doesn't support datums that
+contain more than 10000 files.
+
+`input.atom.from_commit` specifies the starting point of the input branch.  If
+`from_commit` is not specified, then the entire input branch will be
+processed.  Otherwise, only commits since the `from_commit` (not including
+the commit itself) will be processed.
+
+#### Union Input
+
+Union inputs take the union of other inputs. For example:
+
+| inputA | inputB | inputA ∪ inputB |
+| ------ | ------ | --------------- |
+| foo    | fizz   | foo             |
+| bar    | buzz   | fizz            |
+|        |        | bar             |
+|        |        | buzz            |
+
+Notice that union inputs, do not take a name and maintain the names of the sub-inputs.
```

Comment:
Is there a reason union inputs couldn't be (optionally) named? It seems in a lot of cases it would be useful to have the files from any of the atom inputs show up in the same directory so my code doesn't have to manually check multiple places (unless I'm missing I don't think there could be conflicts, right?)